db/compaction_job.cc

```diff
@@ -1265,11 +1267,22 @@ Status CompactionJob::FinishCompactionOutputFile(
         // (the max key in the previous table or subcompaction) in order for
         // files to appear key-space partitioned.
         //
-        // Choose lowest seqnum so this file's smallest internal key comes
-        // after the previous file's/subcompaction's largest. The fake seqnum
-        // is OK because the read path's file-picking code only considers user
-        // key.
-        smallest_candidate = InternalKey(*lower_bound, 0, kTypeRangeDeletion);
+        // When lower_bound is chosen by a subcompaction, we know that
+        // subcompactions over smaller keys cannot contain any keys at
+        // lower_bound. We also know that smaller subcompactions exist, because
+        // otherwise the subcompaction woud be unbounded on the left. As a
+        // result, we know that no other files on the output level will contain
+        // keys at lower_bound. Therefore, it is safe to use the tombstone's
```

Comment:
It is possible though that another output level file's end key is at lower_bound with kMaxSeqnum, right?

Reply:
Yeah good point, that's definitely possible. I'll adjust this to say that "real" keys can't be at lower_bound in other output files.